mgmt/rest/v2/plugin.go

```diff
@@ -226,7 +244,21 @@ func (s *apiV2) loadPlugin(w http.ResponseWriter, r *http.Request, _ httprouter.
 			Write(400, FromError(e), w)
 			return
 		}
+
+		if hasTLS(certPath, keyPath) {
```

Comment:
I have only one small suggestion to add comments to this fragment of code:
- that `hasTLS` checks if one of TLS params (cert or key) has been provided, if not skip the part related to TLS
- that `isTLSEnabled` checks if both of required params have been provided to setup TLS connection; if not return an appropriate error

I made a PR to your repo to not bother you with this small change (https://github.com/candysmurf/snap/pull/4) - use it or add some comments by your own.